app/src/components/common/TmSessionImportName.vue

```diff
@@ -100,18 +106,32 @@ export default {
       },
     },
   },
-  async created() {
-    this.importedAddress = await this.$store.dispatch(`getAddressFromSeed`, {
-      seedPhrase: this.$store.state.recover.seed,
-      network: this.network,
-    })
+  mounted() {
+    this.created()
   },
   methods: {
     onSubmit() {
       this.$v.$touch()
       if (this.$v.name.$invalid) return
       this.$router.push("/recover/password")
     },
+    async created(retry = false) {
+      if (retry) this.attempt++
+      const createAddressResponse = await this.$store.dispatch(
+        `getAddressFromSeed`,
+        {
+          seedPhrase: this.$store.state.recover.seed,
+          network: this.network,
+          // accountType: undefined,
+          attempt: this.attempt,
+        }
+      )
+      this.importedAddress = createAddressResponse.wallet.cosmosAddress
+      this.$store.state.session.accountType =
+        createAddressResponse.wallet.accountType // HACK
+      this.$store.state.session.accountTypeIndex = createAddressResponse.attempt // HACK
```

Comment:
handle locally, doesn't need to be done in store

Reply:
but how? Url params are ugly and the component is not child from this one (navigates to it with the `router`)